providers/v1/barbican/client.go

```diff
@@ -0,0 +1,184 @@
+	allPages, err := secrets.List(c.keyManager, opts).AllPages(ctx)
+	if err != nil {
+		return nil, fmt.Errorf(errClientListAllSecrets, err)
+	}
+
+	allSecrets, err := secrets.ExtractSecrets(allPages)
+	if err != nil {
+		return nil, fmt.Errorf(errClientExtractSecrets, err)
+	}
+
+	if len(allSecrets) == 0 {
+		return nil, fmt.Errorf(errClientGeneric, errors.New("no secrets found"))
+	}
+
+	var secretsMap = make(map[string][]byte)
+
+	// return a secret map with all found secrets.
+	for _, secret := range allSecrets {
+		secretUUID := extractUUIDFromRef(secret.SecretRef)
+		secretsMap[secretUUID], err = secrets.GetPayload(ctx, c.keyManager, secretUUID, nil).Extract()
+		if err != nil {
+			return nil, fmt.Errorf(errClientGetSecretPayload, fmt.Errorf("failed to get secret payload for secret %s: %w", secretUUID, err))
+		}
+	}
+	return secretsMap, nil
+}
+
+// GetSecret retrieves a secret from Barbican.
+func (c *Client) GetSecret(ctx context.Context, ref esapi.ExternalSecretDataRemoteRef) ([]byte, error) {
+	payload, err := secrets.GetPayload(ctx, c.keyManager, ref.Key, nil).Extract()
+	if err != nil {
+		return nil, fmt.Errorf(errClientGetSecretPayload, err)
+	}
+
+	if ref.Property == "" {
+		return payload, nil
+	}
+
+	propertyValue, err := getSecretPayloadProperty(payload, ref.Property)
+	if err != nil {
+		return nil, fmt.Errorf(errClientGetSecretPayloadProperty, fmt.Errorf("failed to get property %s from secret payload: %w", ref.Property, err))
+	}
+
+	return propertyValue, nil
+}
+
+// GetSecretMap retrieves a secret and parses it as a JSON object.
+func (c *Client) GetSecretMap(ctx context.Context, ref esapi.ExternalSecretDataRemoteRef) (map[string][]byte, error) {
+	payload, err := c.GetSecret(ctx, ref)
+	if err != nil {
+		return nil, fmt.Errorf(errClientGeneric, err)
+	}
+
+	var rawJSON map[string]json.RawMessage
+	if err := json.Unmarshal(payload, &rawJSON); err != nil {
+		return nil, fmt.Errorf(errClientJSONUnmarshal, err)
+	}
+
+	secretMap := make(map[string][]byte, len(rawJSON))
+	for k, v := range rawJSON {
+		secretMap[k] = []byte(v)
+	}
+
+	return secretMap, nil
+}
+
+// PushSecret is not implemented right now for Barbican.
+func (c *Client) PushSecret(_ context.Context, _ *corev1.Secret, _ esapi.PushSecretData) error {
+	return fmt.Errorf("barbican provider does not support pushing secrets")
+}
+
+// SecretExists is not implemented right now for Barbican.
+func (c *Client) SecretExists(_ context.Context, _ esapi.PushSecretRemoteRef) (bool, error) {
+	return false, fmt.Errorf("barbican provider does not support checking if a secret exists")
+}
+
+// DeleteSecret is not implemented right now for Barbican.
+func (c *Client) DeleteSecret(_ context.Context, _ esapi.PushSecretRemoteRef) error {
+	return fmt.Errorf("barbican provider does not support deleting secrets")
+}
+
+// Validate checks if the client is properly configured.
+func (c *Client) Validate() (esapi.ValidationResult, error) {
+	return esapi.ValidationResultReady, nil
+}
```

Comment:
```suggestion
// Validate checks if the client is properly configured.
func (c *Client) Validate() (esapi.ValidationResult, error) {
	return esapi.ValidationResultUnknown, nil
}
```

If we have no checks here, we can't really tell this provider is ready. Unknown makes more sense here.